graphql-server/src/utils/servers.ts

```diff
@@ -29,8 +28,14 @@ export function initializeHttpsServer(app: Express) {
 
 export interface ApolloServerContext {
   req: {
-    user: any;
-    isAuthenticated: boolean;
+    user: {
+      email: string;
+      sub: string; // Keycloak user ID
+      groups: Array<string>; // Keycloak user groups
+    };
+    isAuthenticated: () => boolean;
+    logOut: (error: any) => void;
+    app: Express;
   };
   inMemoryCache: NodeCache;
 }
```

Comment:
Added these fields simply for more proper/specific Typescript typing. This does not affect any logic.